### PR TITLE
[ODS-5276] Multiple authorization strategy support for single-item requests

### PR DESCRIPTION
**Pull request by gmcelhanon**
See also: https://github.com/Ed-Fi-Alliance-OSS/Ed-Fi-ODS-Implementation/pull/409

### CODE REVIEW

**Review by bptillman — COMMENTED** [review]
I've reviewed this to the best of my ability and left comments for any of the things I saw. Nothing major stood out and most of my comments revolve around general housekeeping when it comes to cleanup that can be done when removing large chunks of code like what was done here.

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/ViewBasedAuthorizationFilterDefinition.cs [hunk 1]
```diff
@@ -27,30 +32,31 @@ public class ViewFilterApplicationDetails : FilterApplicationDetails
...
+                (criteria, @where, parameters, joinType) => criteria.ApplyJoinFilter(
+                    @where,
+                    parameters,
                     viewName,
                     subjectEndpointName,
                     viewTargetEndpointName,
-                    jt,
-                    Guid.NewGuid().ToString("N")),
+                    joinType,
+                    Guid.NewGuid().ToString("N")), // TODO: GKM - review usage of alias generator here instead of generating GUIDs
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/ViewBasedAuthorizationFilterDefinition.cs:42]: I had intended to go through and review of `TODO: GKM` but forgot to do that. I'll clean this up.

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyFilterDefinitionsFactory.cs [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+
+            IEnumerable<AuthorizationFilterDefinition> CreateEducationOrganizationToSelfPropertyValueFilters()
+            {
+                return _educationOrganizationIdNamesProvider.GetConcreteNames()
+                    .Select(propertyName => 
+                        new AuthorizationFilterDefinition(
+                            $"{propertyName}To{propertyName}",
+                            $"{propertyName} IN (:{propertyName})",
+                            $"{{currentAlias}}.{propertyName} IN (:{propertyName})",
+                            (c, w, p, jt) => w.ApplyPropertyFilters(p, propertyName),
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/RelationshipsAuthorizationStrategyFilterDefinitionsFactory.cs:91]: Clearly they are for `criteria`, `where` (`Junction`), `parameters` and `joinType`. 🙄

I'll change it.

File: Application/EdFi.Ods.Common/Infrastructure/Filtering/AuthorizationFilterDefinition.cs [hunk 3]
```diff
@@ -41,61 +38,22 @@ public class FilterApplicationDetails
         /// <see cref="NHibernate.NHibernateUtil.String" />.  If these conventions don't work for a particular filter definition,
         /// use the other constructor overload.
         /// </remarks>
-        public FilterApplicationDetails(
+        public AuthorizationFilterDefinition(
             string filterName,
-            string friendlyDefaultConditionFormat,
+            string friendlyDefaultConditionFormat, // TODO: GKM - Eliminate if truly unused
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Common/Infrastructure/Filtering/AuthorizationFilterDefinition.cs:43]: This actually did get removed in ODS-5355.

File: Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/ViewBasedAuthorizationFilterDefinition.cs [hunk 4]
```diff
@@ -7,16 +7,21 @@
 using EdFi.Ods.Api.Security.AuthorizationStrategies.NHibernateConfiguration;
 using EdFi.Ods.Common;
 using EdFi.Ods.Common.Infrastructure.Filtering;
+using EdFi.Ods.Common.Security.Authorization;
+using EdFi.Ods.Common.Security.Claims;
+using Microsoft.AspNetCore.Mvc.ViewEngines;
```
[the remark this answers is not in the document]
Comment (gmcelhanon) [reply Application/EdFi.Ods.Api/Security/AuthorizationStrategies/Relationships/Filters/ViewBasedAuthorizationFilterDefinition.cs:12]: Looks like an errant Rider addition based on usage of the class name from another context. Removed.

**Review by bptillman — APPROVED** [review]

**Pull request merged**